### ISSUE TITLE
Automatic boxmap discretization

### ISSUE THREAD

**Issue by cafaxo** [gaioguys/GAIO.jl#31]
It would be really cool if the user could just do
```julia
g = boxmap(f)
```
Once the user applies `g` to a boxset, an automatic/adaptive discretization strategy is used.
A naive approach would be to define an increasing sequence of point discretizations of the unit box and go along this sequence until the image stabilises.
I imagine that this would work well for simple/easy problems.
In this issue, we should decide:
 - Do we want this?
 - If yes, how rigoros should the discretization be?
 - How would it be accessible? (If the automatic strategy is not very good in general, then `boxmap` might not be the most honest way)

This issue depends/extends on #30.

**Comment by gaioguy** [comment]
Yes!  This is something we should definitely do.  There are ideas on an adaptive version in [Junge, O.:
Rigorous discretization of subdivision techniques, Proceedings of EQUADIFF 99, World Scientific, 2000.](http://www-m3.ma.tum.de/twiki/pub/M3/Allgemeines/Publications/Ju99.pdf).

**Comment by cafaxo** [comment]
This method makes sense if `f` is locally Lipschitz.
Do we want to require that for `boxmap(f)`?
If it is locally Lipschirt, how would we determine `L` or would the user have to supply that?

**Comment by gaioguy** [comment]
It seems like a reasonable general assumption to me.  In practice, if one is not aiming for a rigorous computation of a box image (for which I would propose to use interval arithmetic anyway), one could simply evaluate the derivative of the map at the center of the box as an approximation to L - or rather do the SVD of the derivative and construct a properly aligned grid as proposed at the end of the above Ref.

**Comment by cafaxo** [comment]
We could have the case that the user maps a really coarse BoxSet (imagine a single big box) into a really fine partition. Only sampling the center for `L` could be inadequate in this case.

**Comment by gaioguy** [comment]
Right, it would be good to have some way to estimate how well the linearization approximates the map over the box. Maybe sampling at the vertices of the box?
